src/cpp/src/llm_pipeline_static.cpp

```diff
@@ -1036,11 +1057,10 @@ EncodedResults StaticLLMPipeline::generate(
             const auto kv_dim = (output_name.find("value") != std::string::npos &&
                 m_kvcache_desc.v_tensors_transposed) ? 3u : m_kvcache_desc.seq_len;
 
+            auto kvcache_out_tensor = m_kvcache_request.get_tensor(output_name);
             auto kvcache_in_tensor = m_kvcache_request.get_tensor(input_name);
-            auto kvcache_in_slice = make_tensor_slice(
-                kvcache_in_tensor, kv_dim, m_kvcache_desc.num_stored_tokens - 1, m_kvcache_desc.num_stored_tokens
-            );
-            m_kvcache_request.get_tensor(output_name).copy_to(kvcache_in_slice);
+
+            copy_columns_by_row_chunks(kvcache_out_tensor, kvcache_in_tensor, 0, m_kvcache_desc.num_stored_tokens - 1, 1, kv_dim);
```

Comment:
It cannot be done unconditionally as different models may have different `seq_len` position depends on model type and optimization applied

Reply:
I mean, even though it accepts `kv_dim` for some "seq_len" positions, it might be more efficient to use just common copy